Bezier.cpp

```diff
@@ -4,9 +4,70 @@
 Bezier::Bezier(const Vector2& p0, const Vector2& p1, const Vector2& p2, const Vector2& p3)
 	: p0(p0), p1(p1), p2(p2), p3(p3)
 {
+	// Assign points to Bezier
+	this->p0 = p0;
+	this->p1 = p1;
+	this->p2 = p2;
+	this->p3 = p3;
+
+	calculatePoints();
+}
+
+void Bezier::calculatePoints()
+{
+	// Generate curves based on desired smoothness
+	float t = 0;
+	// Angle of line
+	float angle = 0;
+
+	// Generate array of SDL_Point to use in rendering lines
+	for (int i = 0; i < CURVES + 1; i++)
+	{
+		points[i] = Vec2toSDLPoint(getPosAlongBezier(t));
+
+		// Calculate the 
+		if (i > 0)
+		{
+			// Calculate angles for all but the first line of the curve
+			angle = atan2(points[i].y - points[i - 1].y, points[i].x - points[i - 1].x);
+
+			// Calculate corresponding outer and inner points, offset by half of the curve width each
+			pointsOuter[i].x = (WIDTH / 2) * sin(angle) + points[i].x;
+			pointsOuter[i].y = -(WIDTH / 2) * cos(angle) + points[i].y;
+			pointsInner[i].x = -(WIDTH / 2) * sin(angle) + points[i].x;
+			pointsInner[i].y = (WIDTH / 2) * cos(angle) + points[i].y;
+		}
+
+		t += 1.0 / CURVES;
+	}
+
+	// Calculate the angle for the first line of the curve
+	angle = atan2(points[1].y - points[0].y, points[1].x - points[0].x);
+
+	// Calculate the first inner and outer points of the track
+	pointsOuter[0].x = (WIDTH / 2) * sin(angle) + points[0].x;
+	pointsOuter[0].y = -(WIDTH / 2) * cos(angle) + points[0].y;
+	pointsInner[0].x = -(WIDTH / 2) * sin(angle) + points[0].x;
+	pointsInner[0].y = (WIDTH / 2) * cos(angle) + points[0].y;
```

Comment:
You could make this a little more efficient by computing the sine and cosine directly from `points[1] - points[0]` (SOH CAH TOA). In general if you calculate an angle by inverse trig function and then pass it into trig functions, you could have used trig identities instead to avoid calling expensive functions.